python/cudf/cudf/_lib/column.pyx

```diff
@@ -296,13 +313,55 @@ cdef class Column:
         self._children = None
         self._base_children = value
 
+    @property
+    def _is_internally_referenced(self) -> bool:
+        """
+        Determines if any of the buffers underneath the column
+        have been shared internally(i.e., between other columns).
```

Comment:
This isn't quite the right description (only true if copy-on-write is enabled).

---

Ideally, I'd like the `Column` class to be totally agnostic of what kind of `Buffer` lives underneath it.

I wonder if the right solution is for each `Buffer` to expose a `mutable_ptr()` method/attribute. For a regular `Buffer`, that just returns `ptr`. For a `SpillableBuffer`, it spill-locks the buffer before returning `ptr`. For a `CopyOnWriteBuffer`, it unlinks any shared buffers before returning `ptr`.

Would that work? cc: @madsbk

Reply:
After discussion with @shwina made some changes to address this as part of : https://github.com/rapidsai/cudf/pull/11718/commits/bce978108649d82376197937063c7bf1d74fb8e1